packages/socket/lib/cdp-browser.ts

```diff
@@ -0,0 +1,65 @@
+/// <reference lib="dom" />
+import Emitter from 'component-emitter'
+import { v4 as uuidv4 } from 'uuid'
+import { decode, encode } from './utils'
+
+export class CDPBrowserSocket extends Emitter {
+  private _namespace: string
+
+  constructor (namespace: string) {
+    super()
+
+    this._namespace = namespace
+
+    const send = (payload: string) => {
+      // console.log(`[${this._namespace}] send called with`, event, args)
+      const parsed = JSON.parse(payload)
+
+      decode(parsed).then((decoded: any) => {
+        const [event, callbackEvent, args] = decoded
+
+        super.emit(event, ...args)
+        this.emit(callbackEvent, ...args)
+      })
+    }
+
+    // @ts-ignore
+    if (!window[`cypressSocket-${this._namespace}`]) {
+      // @ts-ignore
+      window[`cypressSocket-${this._namespace}`] = {}
+    }
+
+    // @ts-ignore
+    if (!window[`cypressSocket-${this._namespace}`].send) {
+      // @ts-ignore
+      window[`cypressSocket-${this._namespace}`].send = send
+    }
+
+    // Set timeout so that the connect event is emitted after the constructor returns and the user has a chance to attach a listener
+    setTimeout(() => {
+      super.emit('connect')
```

Comment:
do we want to make a connect call to the server to ensure the connection is good before assuming we've connected?

Reply:
We shouldn't need to.  This code gets added with Cypress.  All of the add bindings/scripts/etc. happen prior to any pages loading, so everything should be ready